src/io.md

```diff
@@ -76,3 +76,28 @@ that error explicit.
 
 Note that buffering also works with stdout, so you might want to combine manual
 locking *and* buffering when making many writes to stdout.
+
+## Reading input as raw bytes
+
+The built-in [String](https://doc.rust-lang.org/std/string/struct.String.html) type uses UTF-8
+internally, which adds a small, but nonzero overhead caused by UTF-8 validation when you read
+input into it. If you just want to process input bytes without worrying about UTF-8 (for example if
+you handle ASCII text), you can use the [`read_until`](https://doc.rust-lang.org/std/io/trait.BufRead.html#method.read_until)
+method on `BufReader`:
+```rust
+use std::io::{BufReader, BufRead};
```

Comment:
Does this pass `mdbook test`? I think you might need to wrap this code in a hidden function, see `blah` in the other code examples in this file.